modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/DynamicFilter.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayMultiSelect from '@clayui/multi-select';
+import {fetch, openToast} from 'frontend-js-web';
+import PropTypes from 'prop-types';
+import React, {useEffect, useState} from 'react';
+
+const getSelectedItemsLabel = ({selectedData}) => {
+	return selectedData.map((item) => item.label).join(', ');
+};
+
+const getOdataString = ({id, selectedData}) => {
+	return `${id} in (${selectedData
+		.map((item) => `'${item.value}'`)
+		.join(',')})`;
+};
+
+async function getPicklist(erc) {
+	const response = await fetch(
+		`/o/headless-admin-list-type/v1.0/list-type-definitions/by-external-reference-code/${erc}`
+	);
+
+	if (!response.ok) {
+		openToast({
+			message: Liferay.Language.get('your-request-failed-to-complete'),
+			type: 'danger',
+		});
+
+		return [];
+	}
+
+	return await response.json();
+}
+
+const DynamicFilter = ({
+	id,
+	listTypeDefinitionId,
+	multiple,
+	preselectedValues,
+	selectedData,
```

Comment:
Since this is linked to setFilter, would it make more sense just to pass `filter` in? That way you have `onFilterChange` and `filter`, which lines up more with react verbiage

Reply:
same as above? :thinking: